### PR TITLE
Add wizard-like export view for workflow invocations

### PR DESCRIPTION
**Pull request by davelopez**
This is an attempt to make the workflow invocation export process a bit more guided and user-friendly. It also explores using wizard-like interfaces for "complex" processes with many options.

InvocationWizardFinal

If we like this approach we can make it more generic and incorporate it, for example in the history export (and archival process) or in other places where the user has to decide between multiple options, like collection building, etc.

---

## Generic Wizard Usage
I've extracted the basic logic and the base component into `GenericWizard`. Here is a basic example of how to use it.

**MyWizard.vue**
```vue
<script setup lang="ts">
import { reactive, ref } from "vue";

import { useWizard } from "@/components/Common/Wizard/useWizard";

import GenericWizard from "@/components/Common/Wizard/GenericWizard.vue";

// This reactive object holds the data for the entire wizard.
// Use the values in the wizard steps to control the wizard flow
// and to validate the data before submitting.
const wizardData = reactive({
    enableHiddenStep: false,
    confirmDone: false,
});

// Use the useWizard composable to define the wizard steps
const wizard = useWizard({
    intro: {
        label: "Initial Step",
        instructions: "First instructions go here",
        isValid: () => true,
        isSkippable: () => false,
    },
    "optional-step": {
        label: "Optional",
        instructions: "The instructions for the optional step",
        isValid: () => true,
        isSkippable: () => !wizardData.enableHiddenStep,
    },
    end: {
        label: "Final step",
        instructions: "Mark the checkbox to continue",
        isValid: () => wizardData.confirmDone,
        isSkippable: () => false,
    },
});

const submitButtonLabel = ref("Done");
const isBusy = ref(false);

// This function is called when the wizard is completed
async function onWizardSubmit() {
    isBusy.value = true;
    console.log("Wizard submit...", wizardData);
    await new Promise((resolve) => setTimeout(resolve, 2000));
    console.log("Wizard submit DONE");
    isBusy.value = false;
}
</script>

<template>
    <div>
        <GenericWizard
            title="My Simple Wizard"
            :use="wizard"
            :is-busy="isBusy"
            :submit-button-label="submitButtonLabel"
            @submit="onWizardSubmit">
            <!-- To define the content of each step, use the `v-if` directive with 
            the `wizard.isCurrent` method and pass the corresponding step key -->
            <div v-if="wizard.isCurrent('intro')">
                <p>This is the content of the initial step</p>
                <BFormCheckbox v-model="wizardData.enableHiddenStep" switch> Enable Optional Step </BFormCheckbox>
            </div>

            <div v-if="wizard.isCurrent('optional-step')">
                <p>This is the content of the optional step</p>
                <b>Nothing to do here...</b>
            </div>

            <div v-if="wizard.isCurrent('end')">
                <BFormCheckbox v-model="wizardData.confirmDone">
                    Enable the <b>{{ submitButtonLabel }}</b> button
                </BFormCheckbox>
            </div>
        </GenericWizard>
    </div>
</template>
```

MyWizard

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by nsoranzo** [comment]
Looks great!

**Comment by davelopez** [comment]
This should be now feature-complete and ready for review. I've extracted a generic Wizard component to be reused and included an example in the PR description.

**Comment by itisAliRH** [comment]
Nice work, thanks @davelopez ✌🏻

I did notice a small bug during testing:

https://github.com/user-attachments/assets/0eacff91-479c-4b4f-802d-4a5c11a53d2e

Steps to reproduce:
1- Select `BioCompute Object`, then click "Next".
2- Select `BCO Database`, then click "Next".
3- Return to the first step, select `Research Object Crate`, and click "Next".
4- Notice that the `Temporary Direct Download` option is not automatically selected, yet you can still proceed to the next step.
5- If you proceed, the label on the final button is incorrect.

File: client/src/components/Workflow/Invocation/Export/ExistingInvocationExportProgressCard.vue [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+updateExistingExportProgress();
+
+defineExpose({
+    updateExistingExportProgress,
+});
+</script>
+
+<template>
+    <BCard v-show="hasExistingStsExportData || hasExistingRemoteExportData">
```
Comment (itisAliRH) [inline client/src/components/Workflow/Invocation/Export/ExistingInvocationExportProgressCard.vue]: Minor improvment:
```suggestion
    <BCard v-show="hasExistingStsExportData || hasExistingRemoteExportData" class="mb-2">
```
|Before|After|
|---|----|
|Screenshot From 2024-09-17 15-13-18|Screenshot From 2024-09-17 15-12-58|

**Comment by davelopez** [comment]
Thanks @stain, is https://github.com/galaxyproject/galaxy/pull/18796/commits/c9bea5afcaf85764083724b4519b23d5fe627db3 what you mean?

**Review by bgruening — APPROVED** [review]

**Review by itisAliRH — APPROVED** [review]

**Pull request merged**